### PR TITLE
[BUG Fixed]Fixed hard coding of roles regarding user and admin.

### PR DESCRIPTION
**Pull request by preranah7**
## 📋 Description

This PR fixes the hardcoding of roles by defining a file named constant.js and variables in it.
Those const variables are used in other files where role check is needed.

## 🔨 Changes Made
- Changed user model defined enum type
- Added a constant.js file
- defined const variables in constant.js
- made changes accordingly in other files.

## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

- [✅ ] My code follows the style guidelines of this project.
- [✅] I have performed a self-review of my own code.
- [✅ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [✅ ] My changes generate no new warnings.
- [✅ ] I have tested it locally and it works fine.
- [ ] Any dependent changes have been merged and published in downstream modules.

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [✅ ] Bug fix (non-breaking change which fixes an issue) 🐛
- [ ] New feature (non-breaking change which adds functionality) ✨
- [ ] UI enhancement (non-breaking change which enhances UI) 🎨
- [ ] Documentation update 📚

## Reminder:
Please ensure your pull request is made to the `test` branch, not `main`.

### CODE REVIEW

File: utils/constant.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const userRole = 'user'
+const adminRole = 'admin'
+module.exports = {userRole,adminRole}
```
Comment (Sowmiyajayakumar) [inline utils/constant.js:3]: variables in Constant.js should be in uppercase like ROLE_ADMIN, ROLE_USER

File: controllers/campgrounds.js [hunk 2]
```diff
@@ -93,7 +94,7 @@ module.exports.showCampground = async (req, res) => {
         req.flash('error','Cannot find that campground!');
         return res.redirect('/campgrounds');
     }
-    res.render('campgrounds/show',{campground});
+    res.render('campgrounds/show',{campground,userRole,adminRole});
```
Comment (Sowmiyajayakumar) [inline controllers/campgrounds.js:97]: This change is not required

File: utils/constant.js (hunk 1 above)
Comment (preranah7) [reply utils/constant.js:3]: Sure, I will make the changes accordingly.

File: controllers/campgrounds.js (hunk 2 above)
Comment (preranah7) [reply controllers/campgrounds.js:97]: But the show.ejs file is giving error and not rendering the page saying userRole , adminRole not defined.
Hence I passed those variables and it is getting rendered without giving any error. 
Should I remove it? 
Screenshot (2721)
Kindly review the error and guide me if I should remove those passed variables?

File: models/user.js [hunk 3]
```diff
@@ -9,7 +9,8 @@ const UserSchema= new Schema({
     },
     role:{
         type:String,
-        default:"user"
+        enum: ['user', 'admin'],  // Define enum for the role field
+        default: 'user'           // Set default role to 'user'
```
Comment (Sowmiyajayakumar) [inline models/user.js:13]: use the constant variables here as well

File: middleware.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 const {campgroundSchema, reviewSchema}= require('./schemas.js');
 const ExpressError=require('./utils/ExpressError');
+const {ROLE_USER , ROLE_ADMIN} = require('../utils/constant');
```
Comment (Sowmiyajayakumar) [inline middleware.js:3]: path is wrong. this is the right path './utils/constant'

**Comment by preranah7** [comment]
I will update it.

**Comment by Sowmiyajayakumar** [comment]
Hey @preranah7 , are you working on this?

**Comment by preranah7** [comment]
Yes, I will submit Pull Request soon.

**Pull request closed without merging**